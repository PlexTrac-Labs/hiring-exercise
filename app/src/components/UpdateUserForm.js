import React, { useRef, useState, useEffect } from "react";
import axios from "axios";

export default function UpdateUserForm({ user }) {
  const firstNameRef = useRef();
  const lastNameRef = useRef();
  const usernameRef = useRef();
  const emailRef = useRef();
  const birthYearRef = useRef();
  const favColorRef = useRef();

  function handleUpdateUserSubmit(e) {
    e.preventDefault();
    const firstName = firstNameRef.current.value;
    const lastName = lastNameRef.current.value;
    const username = usernameRef.current.value;
    const email = emailRef.current.value;
    const birthYear = birthYearRef.current.value;
    const favColor = favColorRef.current.value;
    axios
      .put(`/user/${user._id}`, {
        firstName: firstName,
        lastName: lastName,
        username: username,
        email: email,
        birthYear: birthYear,
        favColor: favColor
      })
      .then(response => {
        window.location = "/profile";
      });
  }

  return (
    <form className="updateUser" onSubmit={handleUpdateUserSubmit}>
      <div>
        <input
          id="currentFirstNameInput"
          ref={firstNameRef}
          placeholder={user.firstName}
          type="text"
          name="firstName"
        />
        <label htmlFor="currentFirstNameInput">First Name</label>
      </div>
      <div>
        <input
          id="currentLastNameInput"
          ref={lastNameRef}
          placeholder={user.lastName}
          type="text"
          name="lastName"
        />
        <label htmlFor="currentLastNameInput">Last Name</label>
      </div>
      <div>
        <input
          id="currentUserNameInput"
          ref={usernameRef}
          placeholder={user.username}
          type="text"
          name="username"
        />
        <label htmlFor="currentUserNameInput">User Name</label>
      </div>
      <div>
        <input
          id="currentEmailInput"
          ref={emailRef}
          placeholder={user.email}
          type="text"
          name="email"
        />
        <label htmlFor="currentEmailInput">Email</label>
      </div>
      <div>
        <input
          id="currentBirthYearInput"
          ref={birthYearRef}
          placeholder={user.birthYear}
          type="text"
          name="birthYear"
        />
        <label htmlFor="currentBirthYearInput">Year of Birth</label>
      </div>
      <div>
        <input
          id="currentFavColorInput"
          ref={favColorRef}
          placeholder={user.favColor}
          type="text"
          name="favColor"
        />
        <label htmlFor="currentFavColorInput">Favorite Color</label>
      </div>
      <button className="w-100 btn btn-lg btn-primary" type="submit">
        Submit
      </button>
    </form>
  );
}
