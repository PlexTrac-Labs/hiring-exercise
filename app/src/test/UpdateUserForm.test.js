import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import UpdateUserForm from "../components/UpdateUserForm";

//Create user to use for testing
const user = {
  firstName: "John",
  lastName: "Doe",
  username: "jdoe",
  email: "johndoe@mail.com",
  birthYear: "1999",
  favColor: "Blue"
};

//Test to make sure the App, entry point to everything works.
it("renders correctly", async () => {
  const tree = renderer.create(<UpdateUserForm user={user} />).toJSON();
  expect(tree).toMatchSnapshot();
});
