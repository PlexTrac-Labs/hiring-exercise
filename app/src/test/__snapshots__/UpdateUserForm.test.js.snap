// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="updateUser"
  onSubmit={[Function]}
>
  <div>
    <input
      id="currentFirstNameInput"
      name="firstName"
      placeholder="John"
      type="text"
    />
    <label
      htmlFor="currentFirstNameInput"
    >
      First Name
    </label>
  </div>
  <div>
    <input
      id="currentLastNameInput"
      name="lastName"
      placeholder="Doe"
      type="text"
    />
    <label
      htmlFor="currentLastNameInput"
    >
      Last Name
    </label>
  </div>
  <div>
    <input
      id="currentUserNameInput"
      name="username"
      placeholder="jdoe"
      type="text"
    />
    <label
      htmlFor="currentUserNameInput"
    >
      User Name
    </label>
  </div>
  <div>
    <input
      id="currentEmailInput"
      name="email"
      placeholder="johndoe@mail.com"
      type="text"
    />
    <label
      htmlFor="currentEmailInput"
    >
      Email
    </label>
  </div>
  <div>
    <input
      id="currentBirthYearInput"
      name="birthYear"
      placeholder="1999"
      type="text"
    />
    <label
      htmlFor="currentBirthYearInput"
    >
      Year of Birth
    </label>
  </div>
  <div>
    <input
      id="currentFavColorInput"
      name="favColor"
      placeholder="Blue"
      type="text"
    />
    <label
      htmlFor="currentFavColorInput"
    >
      Favorite Color
    </label>
  </div>
  <button
    className="w-100 btn btn-lg btn-primary"
    type="submit"
  >
    Submit
  </button>
</form>
`;
